apis/sgv2-quarkus-common/src/main/java/io/stargate/sgv2/api/common/grpc/StargateBridgeInterceptor.java

```diff
@@ -0,0 +1,34 @@
+package io.stargate.sgv2.api.common.grpc;
+
+import io.grpc.CallOptions;
+import io.grpc.Channel;
+import io.grpc.ClientCall;
+import io.grpc.ClientInterceptor;
+import io.grpc.Metadata;
+import io.grpc.MethodDescriptor;
+import io.quarkus.grpc.GlobalInterceptor;
+import io.stargate.sgv2.api.common.StargateRequestInfo;
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+@GlobalInterceptor
+@ApplicationScoped
+public class StargateBridgeInterceptor implements ClientInterceptor {
+  @Inject StargateRequestInfo requestInfo;
+  @Inject GrpcClients grpcClients;
+
+  @Override
+  public <ReqT, RespT> ClientCall<ReqT, RespT> interceptCall(
+      MethodDescriptor<ReqT, RespT> method, CallOptions callOptions, Channel next) {
+    Metadata.Key<String> tenantIdKey = grpcClients.getTenantIdKey();
+    Metadata.Key<String> cassandraTokenKey = grpcClients.getCassandraTokenKey();
+    return next.newCall(
+        method,
+        callOptions.withCallCredentials(
+            new StargateRequestCallCredentials(
+                tenantIdKey,
+                cassandraTokenKey,
+                requestInfo.getTenantId(),
+                requestInfo.getCassandraToken())));
+  }
```

Comment:
wait why just not add the metadata, there is a call example in grpc itself

```
 private static final class HeaderAttachingClientInterceptor implements ClientInterceptor {
        private final Metadata extraHeaders;

        HeaderAttachingClientInterceptor(Metadata extraHeaders) {
            this.extraHeaders = (Metadata)Preconditions.checkNotNull(extraHeaders, "extraHeaders");
        }

        public <ReqT, RespT> ClientCall<ReqT, RespT> interceptCall(MethodDescriptor<ReqT, RespT> method, CallOptions callOptions, Channel next) {
            return new HeaderAttachingClientCall(next.newCall(method, callOptions));
        }

        private final class HeaderAttachingClientCall<ReqT, RespT> extends ForwardingClientCall.SimpleForwardingClientCall<ReqT, RespT> {
            HeaderAttachingClientCall(ClientCall<ReqT, RespT> call) {
                super(call);
            }

            public void start(ClientCall.Listener<RespT> responseListener, Metadata headers) {
                headers.merge(HeaderAttachingClientInterceptor.this.extraHeaders);
                super.start(responseListener, headers);
            }
        }
    }
```

Reply:
you can simply copy the `HeaderAttachingClientCall` and enable passing of the `extraHeaders` via constuctor, makes it static as well :) The interceptor can then calculate metadata as in `GrpcClients` and just pass on..